src/libs/actions/IOU.ts

```diff
@@ -1194,6 +1198,72 @@ function getFieldViolationsOnyxData(iouReport: OnyxTypes.Report): SetRequired<On
     };
 }
 
+function buildOnyxDataForTestDriveIOU(
+    transaction: OnyxTypes.Transaction,
+    iouOptimisticParams: MoneyRequestOptimisticParams['iou'],
+    chatOptimisticParams: MoneyRequestOptimisticParams['chat'],
+): [OnyxUpdate[], OnyxUpdate[], OnyxUpdate[]] {
+    const optimisticData: OnyxUpdate[] = [];
+    const successData: OnyxUpdate[] = [];
+    const failureData: OnyxUpdate[] = [];
+
+    const optimisticIOUReportAction = buildOptimisticIOUReportAction({
+        type: CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        comment: '',
+        participants: transaction.participants ?? [],
+        paymentType: CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
+        iouReportID: iouOptimisticParams.report.reportID,
+        transactionID: transaction.transactionID,
+    });
+
+    const text = Localize.translateLocal('testDrive.employeeInviteMessage', {name: getDisplayNameOrDefault(personalDetailsList?.[userAccountID])});
+    const textComment = buildOptimisticAddCommentReportAction(Parser.htmlToMarkdown(text), undefined, userAccountID);
+    const iouCreatedActionDate = parse(iouOptimisticParams.createdAction.created, CONST.DATE.FNS_DB_FORMAT_STRING, new Date());
+    const commentReportActionDate = subMilliseconds(iouCreatedActionDate, 1);
+    textComment.reportAction.created = format(commentReportActionDate, CONST.DATE.FNS_DB_FORMAT_STRING);
+
+    optimisticData.push(
+        // TODO: For some reason this message is not being persisted/processed in the BE.
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatOptimisticParams.report?.reportID}`,
+            value: {
+                [textComment.reportAction.reportActionID]: textComment.reportAction as ReportAction,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${iouOptimisticParams.report.reportID}`,
+            value: {
+                ...iouOptimisticParams.report,
+                ...{lastActionType: CONST.REPORT.ACTIONS.TYPE.MARKED_REIMBURSED, statusNum: CONST.REPORT.STATUS_NUM.REIMBURSED},
+                hasOutstandingChildRequest: false,
+                lastActorAccountID: currentUserPersonalDetails?.accountID,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${iouOptimisticParams.report.reportID}`,
+            value: {
+                [iouOptimisticParams.action.reportActionID]: {
+                    ...(optimisticIOUReportAction as OnyxTypes.ReportAction),
+                },
+            },
```

Comment:
```suggestion
            value: {
                [iouOptimisticParams.action.reportActionID]: optimisticIOUReportAction,
            },
```

No need for shallow copy